lib/chewy/type/crutch.rb

```diff
@@ -15,7 +15,12 @@ def initialize(type, collection)
           @type._crutches.each_key do |name|
             singleton_class.class_eval <<-METHOD, __FILE__, __LINE__ + 1
               def #{name}
-                @#{name} ||= @type._crutches[:#{name}].call @collection
+                execution_block = @type._crutches[:#{name}]
+                if execution_block.arity == 1
+                  @#{name} ||= execution_block.call @collection
+                else
+                  @#{name} ||= execution_block.call @collection, self
```

Comment:
will this be ok when the crutch hasn't been loaded yet? like say `accounts` gets initialized before `contacts` ?

Reply:
looks like technically it will be OK bc this is just a proc

Reply:
Yes. It would work, because crutches are defined at class load time, but executed at import time.

and definition of these functions doesnt depend on each other.

Reply:
This can clobber methods with the same name in the crutches object right? We probably should have an rspec that flags certain crutch names.